nf_core/download.py

```diff
@@ -838,81 +847,107 @@ def rectify_raw_container_matches(self, raw_findings):
-                    pattern = re.escape(container_definition.group(1))
-                    # extract the quoted string(s) following the variable assignment
-                    container_names = re.findall(r"%s\s*=\s*[\"\']([^\"\']+)[\"\']" % pattern, search_space)
-
-                    if bool(container_names):
-                        if isinstance(container_names, str):
-                            this_container = f"https://depot.galaxyproject.org/singularity/{container_names}"
-
-                        elif isinstance(container_names, list):
-                            # this deliberately appends container_names[-1] twice to cleaned_matches
-                            # but deduplication is performed anyway and just setting this_container
-                            # here as well allows for an easy check to see if parsing succeeded.
-                            for container_name in container_names:
-                                this_container = f"https://depot.galaxyproject.org/singularity/{container_name}"
-                                cleaned_matches.append(this_container)
-                        else:
-                            # didn't find valid container declaration, but parsing succeeded.
-                            this_container = None
+        Conceptually, this works like so:
+
+        Everything after the last Slash should be identical, e.g. "scanpy:1.7.2--pyhdfd78af_0" in
+        ['https://depot.galaxyproject.org/singularity/scanpy:1.7.2--pyhdfd78af_0', 'biocontainers/scanpy:1.7.2--pyhdfd78af_0']
 
-            # If we have a this_container, parsing succeeded.
-            if this_container:
-                cleaned_matches.append(this_container)
-            else:
-                log.error(
-                    f"[red]Cannot parse container string in '{file_path}':\n\n{textwrap.indent(container_value, '    ')}\n\n:warning: Skipping this singularity image."
-                )
 
-        return cleaned_matches
+        re.sub('.*/(.*)','\\1',c) will drop everything up to the last slash from c (container_id)
+
+        d.get(k:=re.sub('.*/(.*)','\\1',c),'') assigns the truncated string to k (key) and gets the
+        corresponding value from the dict if present or else defaults to "".
+
+        If the regex pattern matches, the original container_id will be assigned to the dict with the k key.
+        r"^$|(?!^http)" matches an empty string (we didn't have it in the dict yet and want to keep it in either case) or
+        any string that does not start with http. Because if our current dict value already starts with http,
+        we want to keep it and not replace with with whatever we have now (which might be the Docker URI).
+
+        A regex that matches http, r"^$|^http" could thus be used to prioritize the Docker URIs over http Downloads
+        """
+        d = {}
+        for c in container_list:
+            log.info(c)
```

Comment:
Can we add a message? For example:
```suggestion
            log.info(f"Evaluating container {c}")
```